### PR TITLE
Download link activity page

### PR DESCRIPTION
**Pull request by servingUpAces**
We removed the download link on the activity page (click the big numbers) because it would time out for large requests. 
When we fix the download functionality we didn't put the link back on the page. This PR adds the link back.

Some refactoring is required as there is some replication of code, but I wanted to get this out there and tested. We may also want to iterate on the columns returned in the CSV.

### CODE REVIEW

File: app/main/views/jobs.py [hunk 1]
```diff
@@ -35,39 +32,11 @@
     user_has_permissions,
     generate_notifications_csv,
     get_time_left,
-    REQUESTED_STATUSES,
-    FAILURE_STATUSES,
-    SENDING_STATUSES,
-    DELIVERED_STATUSES,
     get_letter_timings,
-)
+    parse_filter_args, set_status_filters)
```
Comment (quis) [inline app/main/views/jobs.py]: I’d prefer bracket on a new line and trailing comma like this:
```python
    parse_filter_args,
    set_status_filters,
)
```
makes for cleaner diffs in the future

File: app/utils.py [hunk 2]
```diff
@@ -132,21 +135,36 @@ def generate_notifications_csv(**kwargs):
...
+                    notification['status'],
+                    notification['created_at']
+                ]
+                line = ','.join(str(i) for i in values) + '\n'
+                yield line
+        else:
+            # Change here
+            for notification in notifications:
+                values = [
+                    notification.get('row_number', None),
```
Comment (quis) [inline app/utils.py]: I think we should lose this column, doesn’t make too much sense in the all notifications report.

File: app/utils.py [hunk 3]
```diff
@@ -132,21 +135,36 @@ def generate_notifications_csv(**kwargs):
...
+                    notification['to'],
+                    notification['template']['name'],
+                    notification['template']['template_type'],
+                    notification.get('job_name', None),
+                    notification['status'],
+                    notification['created_at']
```
Comment (quis) [inline app/utils.py]: Should have `updated_at` as a column here too – need to think of a sensible column name for it.

Comment (servingUpAces) [reply app/utils.py]: We use "completed_at" for get_notifications

**Pull request merged**